### PR TITLE
Adds Sentry for Error handling

### PR DESCRIPTION
**Pull request by orta**
This adds Sentry, replacing Hockey for crash reporting, as we're standardising on Sentry across Artsy. 

* Adds the Sentry Pod, this is their main supported one, but they do have a non-Swift version too
* Adds new CocoaPods Keys, these can be found in the team 1password
* Adds a new analytical provider (in the app, the one in ARAnayltics is for the old SDK)
* Adds uploading dSYMs to Sentry during deployment

TODO because I want to give it a final look over tomorrow.

Might need to do this too: https://blog.sentry.io/2017/05/01/release-commits.html

### CODE REVIEW

File: Artsy/App/ARAppDelegate+Analytics.m [hunk 1]
```diff
@@ -89,22 +90,25 @@ - (void)setupAnalytics
...
         ARAnalyticsVisualizer *visualizer = [ARAnalyticsVisualizer new];
         [ARAnalytics setupProvider:visualizer];
     }
 
-#if DEBUG
+    id sentry = [[ARSentryAnalyticsProvider alloc] initWithDSN:sentryEnv];
+    [ARAnalytics setupProvider:sentry];
+
     BITHockeyManager *hockey = [BITHockeyManager sharedHockeyManager];
-    hockey.disableUpdateManager = YES;
```
Comment (orta) [inline Artsy/App/ARAppDelegate+Analytics.m]: updates and crashes are now disabled from Hockey

File: Artsy/App/ARSentryAnalyticsProvider.m [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    [SentryClient shared].user = [[SentryUser alloc] initWithId:userID email:email username:email extra:@{}];
+}
+
+// An event in Sentry terms is a crash, so in this case we're sending along events from analytics as breadcrumbs
+
+- (void)event:(NSString *)event withProperties:(NSDictionary *)properties
+{
+    SentryBreadcrumb *breadcrumb = [[SentryBreadcrumb alloc] initWithCategory:event timestamp:[NSDate new] message:nil type:nil level:SentrySeverityDebug data:properties];
+    [[SentryClient shared].breadcrumbs add:breadcrumb];
+}
```
Comment (orta) [inline Artsy/App/ARSentryAnalyticsProvider.m]: This should be 👍

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Artsy/App/ARSentryAnalyticsProvider.m [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+@import Sentry;
+
+@implementation ARSentryAnalyticsProvider
+
+- (id)initWithDSN:(NSString *)DSN
+{
+    self = [super init];
+    if (!self) { return nil; }
+    SentryClient *client = [[SentryClient alloc] initWithDsnString:DSN];
+    [SentryClient setShared: client];
```
Comment (alloy) [inline Artsy/App/ARSentryAnalyticsProvider.m]: Oh the horror of that stray space 🙈

File: Podfile.lock [hunk 4]
```diff
@@ -186,6 +249,8 @@ PODS:
   - SDWebImage (3.7.3):
     - SDWebImage/Core (= 3.7.3)
   - SDWebImage/Core (3.7.3)
+  - Sentry (2.1.9):
+    - KSCrash (~> 1.15.5)
```
Comment (alloy) [inline Podfile.lock]: KSCrash looks pretty sweet https://github.com/kstenerud/KSCrash

File: fastlane/Fastfile [hunk 5]
```diff
@@ -58,6 +58,15 @@ lane :ship_beta do
...
          repository_url: "https://github.com/artsy/eigen.git", commit_sha: commit
 
+  upload_symbols_to_sentry auth_token: ENV['SentryUploadAuthKey'],
+                           org_slug: 'artsynet',
+                           project_slug: 'eigen'
+
+
+  upload_symbols_to_sentry auth_token: ENV['SentryUploadAuthKey'],
+                           org_slug: 'artsynet',
+                           project_slug: 'eigen-staging '
```
Comment (alloy) [inline fastlane/Fastfile]: Do we _need_ different envs or is this done for our own convenience of separating reports from beta testers vs production?

**Comment by alloy** [comment]
That release commits feature looks nice too 👌

File: fastlane/Fastfile (hunk 5 above)
Comment (orta) [inline fastlane/Fastfile]: Separation of concerns and consistency mainly

Comment (alloy) [inline fastlane/Fastfile]: Need to also upload `Pods/Emission/Assets/Emission.js.map`

**Comment by alloy** [comment]
So how do we also enable crash reporting from RN? (Besides also needing to upload the sourcemaps)

**Comment by orta** [comment]
There's a separate lib that we'd need to ship - https://github.com/getsentry/react-native-sentry

File: fastlane/Fastfile (hunk 5 above)
Comment (orta) [inline fastlane/Fastfile]: 👍

**Comment by alloy** [comment]
Ok, I’ll add that to Emission 👍

**Comment by orta** [comment]
OK, I've gone through the docs and everything seems 👍 - I'm going to crash the app and verify everything looks good, if it works out then this is 🍏

**Comment by orta** [comment]
A crash triggers:

```
> 2017-05-11 12:43:31.759 Artsy[15197:32841116] 	[Adjust]w: Unable to read iAd details
2017-05-11 12:43:33.759 Artsy[15197:32841271] 	[Adjust]w: Reached limit number of retry for iAd v3. Trying iAd v2
fatal error: TEST - Sentry Client Crash: file /Users/orta/dev/projects/artsy/ios/eigen/Pods/Sentry/Sources/Sentry.swift, line 217
INFO : KSCrashReport.c (1709): void kscrashreport_writeStandardReport(const KSCrash_MonitorContext *const, const char *const): Writing crash report to /Users/orta/Library/Developer/CoreSimulator/Devices/E7CF0021-3BB5-4999-9022-D8E21A92462F/data/Containers/Data/Application/2488737D-BC35-4C7D-AC15-A5F7D8196A9A/Library/Caches/KSCrash/Artsy/Reports/Artsy-report-000000004f000000.json
ERROR: KSMachineContext.c (232): void ksmc_resumeEnvironment(): thread_resume (ffffffff): (ipc/send) invalid destination port
```

Which causes: https://sentry.io/artsynet/eigen-staging/issues/271169511/

**Comment by orta** [comment]
Actually, I can get these releases in, give me 10m

**Comment by orta** [comment]
I've got in the release API, but it's inelegant - if you wanna improve it, that's your call (and probably super easy) - I ran out of time

**Comment by alloy** [comment]
@orta I’m not seeing any dSYM bundles uploaded https://sentry.io/artsynet/eigen-staging/settings/debug-symbols/ ? I also got an email from Sentry:

screen shot 2017-05-11 at 14 08 46

**Comment by orta** [comment]
it can't because we upload on deployment

**Comment by orta** [comment]
that was me running it in a sim, which shouldn't happen normally because there'll be a debugger attached to handle the crash

**Comment by alloy** [comment]
Aha. Nonetheless, are you sure it will upload all the dSYMs?

I also wonder why [the fastlane plugin](https://github.com/fastlane/fastlane/blob/22e2c76de7d1110651ad8d8c40bc276ffe2c461b/fastlane/lib/fastlane/actions/upload_symbols_to_sentry.rb#L6-L8) recommends that we use [the official plugin](https://github.com/getsentry/fastlane-plugin-sentry#sentry-actions) instead?

**Comment by orta** [comment]
Perfect

**Comment by alloy** [comment]
Ahhhh, don't think I would've found that :) Did you kickoff another beta test?

**Comment by orta** [comment]
I'm not 100% - I did see it in the troubleshooting, just sent off a beta from it though, so we'll see

File: fastlane/Fastfile [hunk 6]
```diff
@@ -58,9 +58,56 @@ lane :ship_beta do
...
+    "refs": [\{
+        "repository":"artsy/eigen",
+        "commit":"#{commit}"
+    \}],
+    "projects":["eigen","eigen-staging"]
+  \}
+  '
+  }
+
+ `curl https://sentry.io/api/0/organizations/artsynet/releases/ -X POST  -H 'Authorization: Bearer {TOKEN}' -H 'Content-Type: application/json' #{sentry_json}`
```
Comment (alloy) [inline fastlane/Fastfile]: Is it ok  that this now happens after creating the release?

Comment (orta) [inline fastlane/Fastfile]: Yeah, in theory they all should pass so it shouldn't break :)

**Comment by alloy** [comment]
HockeyApp uploads are now failing, but do we really still need those?

**Comment by orta** [comment]
The error is that there are problems with uploading to Hockey, 

```
Error when trying to upload file(s) to HockeyApp: 504 - <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
<TITLE>ERROR: The request could not be satisfied</TITLE>
</HEAD><BODY>
<H1>ERROR</H1>
<H2>The request could not be satisfied.</H2>
<HR noshade size="1px">
CloudFront attempted to establish a connection with the origin, but either the attempt failed or the origin closed the connection.
<BR clear="all">
<HR noshade size="1px">
<PRE>
Generated by cloudfront (CloudFront)
Request ID: cnihryfFwJn5Ivvc4QleoV7IdSRnyiS__RrB164tEYvOpuVKuZLk0w==
</PRE>
<ADDRESS>
</ADDRESS>
</BODY></HTML>
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/ui/interface.rb:154:in `user_error!'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/ui/ui.rb:14:in `method_missing'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/actions/hockey.rb:184:in `run'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:252:in `block (2 levels) in execute_action'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:230:in `block in execute_action'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:226:in `chdir'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:226:in `execute_action'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
	Fastfile:52:in `block in parsing_binding'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/lane.rb:33:in `call'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:45:in `execute'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/lane_manager.rb:52:in `cruise_lane'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/commands_generator.rb:104:in `block (2 levels) in run'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.4/lib/commander/command.rb:178:in `call'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.4/lib/commander/command.rb:153:in `run'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.4/lib/commander/runner.rb:476:in `run_active_command'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:40:in `run!'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.4/lib/commander/delegates.rb:15:in `run!'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/commands_generator.rb:303:in `run'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/cli_tools_distributor.rb:66:in `take_off'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/bin/fastlane:20:in `<top (required)>'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/bin/fastlane:23:in `load'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/bin/fastlane:23:in `<top (required)>'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/cli/exec.rb:74:in `load'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/cli/exec.rb:74:in `kernel_load'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/cli/exec.rb:27:in `run'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/cli.rb:332:in `exec'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/cli.rb:20:in `dispatch'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/cli.rb:11:in `start'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/exe/bundle:34:in `block in <top (required)>'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/friendly_errors.rb:100:in `with_friendly_errors'
	/opt/rubies/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/exe/bundle:26:in `<top (required)>'
	/opt/rubies/ruby-2.3.1/bin/bundle:23:in `load'
	/opt/rubies/ruby-2.3.1/bin/bundle:23:in `<main>'
```

Perhaps their's been an API breaking change? https://github.com/fastlane/fastlane/issues/9105

**Comment by orta** [comment]
Hrm, I'm struggling to replicate the code signed setup locally - it definitely looks like it's going through the list of dSYMs and uploading them individually - the _implied_ issue is that one specific upload is over 20 (or maybe 40) MB big

**Comment by orta** [comment]
Yep, Eigen is 22MB - seeing what I can do to get the higher limit

screen shot 2017-05-15 at 15 44 31

**Comment by orta** [comment]
In [their forums](https://forum.sentry.io/t/413-error-when-uploading-bundle-js-map/297/6) they mention updating the limit to 20MB - the [last log](https://circleci.com/api/v1.1/project/github/artsy/eigen/4148/output/22/0?file=true) wasn't perfectly conducive to getting the error to say that this was the problem

**Comment by orta** [comment]
I made a fork of the fastlane plugin with less verbose verbose output :)

**Comment by orta** [comment]
Looks [like it's batching all 61 dSYMs into one build](https://circleci.com/gh/artsy/eigen/4156?utm_campaign=chatroom-integration&utm_medium=referral&utm_source=slack) inside the sentry-cli 

```
INFO [2017-05-15 09:26:01.28]: $ sentry-cli upload-dsym '/Users/distiller/eigen/Artsy.app.dSYM.zip'
DEBUG [2017-05-15 09:26:01.28]: 

INFO [2017-05-15 09:27:18.85]: Batch 1
INFO [2017-05-15 09:27:18.85]: [1/3] Found 61 debug symbol files. Checking for missing symbols on server
INFO [2017-05-15 09:27:18.85]: [2/3] Compressing 61 missing debug symbol files
INFO [2017-05-15 09:27:18.85]: [3/3] Uploading debug symbol files
```

**Comment by alloy** [comment]
> Hrm, I'm struggling to replicate the code signed setup locally

You need to import the pkey belonging to the cert from the mobile provisioning repo into your own keychain.

**Comment by orta** [comment]
Going to look at using the last release of the fastlane plugin before they moved to using `sentry-cli` - it's in Rust, which is a toolchain I don't know yet, and making a fix is probably a biggish task due to that

**Comment by orta** [comment]
Him, the [older build only has the dsym](https://github.com/getsentry/fastlane-plugin-sentry/releases) (see 0.1.2) option. I'm gonna try it quickly to test.

**Comment by orta** [comment]
The [older build did not work](https://circleci.com/gh/artsy/eigen/4161?utm_campaign=chatroom-integration&utm_medium=referral&utm_source=slack), same 413 - and it leaked our API token, which I've cycled :D

Kinda have two options right now, we write some ruby to push each dsym individually or look into figuring out how the batching works inside sentry-cli

**Comment by orta** [comment]
Had another idea, the dsyms are currently zipped up. If we unzip it on the server and try passing the folder of the unzipped dsyms it may make the batch work in our favour.

**Comment by mitsuhiko** [comment]
@orta the current upload limits are 150MB after zipping for debug symbols (which is why so many are contained in one upload) and ~40MB for JS assets in case you have react-native which are uploaded individually at the moment.

In case you do get errors please run with `--log-level=debug` so we can figure out what's happening.

**Comment by mitsuhiko** [comment]
The constant in sentry-cli for this is here if it needs tweaking: https://github.com/getsentry/sentry-cli/blob/4533c72ef0236350441702760131314f95e48d1f/src/commands/upload_dsym.rs#L28-L31

**Comment by orta** [comment]
Thanks @mitsuhiko - that helps! - we [do _have_ some logs](https://circleci.com/api/v1.1/project/github/artsy/eigen/4148/output/22/0?file=true) with the `--log-level=debug ` but they're too long and Circle crops them :D. Tomorrow I'll pipe them to an external file and get them accessible.

**Comment by mitsuhiko** [comment]
Ugh. So much useless output in that file. Need to figure out why we dump out so much (eg: nobody cares about the macho parser). And then it ends with 

`This output was truncated to stay under the 4MB limit.` :D 

Will fix that so it spits out less at that level.

File: fastlane/Fastfile [hunk 7]
```diff
@@ -48,17 +48,55 @@ lane :ship_beta do
...
+    end
+
+
+    puts "Uploaded dsym for #{project_slug}"
+
+    sentry_upload_file auth_token: ENV['SentryUploadAuthKey'],
+                       org_slug: org_slug,
+                       project_slug: project_slug,
+                       version: latest_version,
+                       dist: build_version,
```
Comment (orta) [inline fastlane/Fastfile]: this build_version needs to be a string:

```
'dist' value must be a String! Found Fixnum instead.
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/ui/interface.rb:154:in `user_error!'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/ui/ui.rb:14:in `method_missing'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/configuration/config_item.rb:118:in `valid?'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/configuration/config_item.rb:107:in `verify!'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/configuration/configuration.rb:82:in `block in verify_value_exists'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/configuration/configuration.rb:76:in `each'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/configuration/configuration.rb:76:in `verify_value_exists'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/configuration/configuration.rb:60:in `initialize'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/configuration/configuration.rb:31:in `new'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane_core/lib/fastlane_core/configuration/configuration.rb:31:in `create'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/configuration_helper.rb:8:in `parse'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:236:in `block (2 levels) in execute_action'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:230:in `block in execute_action'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:226:in `chdir'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:226:in `execute_action'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
	/Users/distiller/eigen/vendor/bundle/ruby/2.3.0/gems/fastlane-2.30.2/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
	Fastfile:71:in `block (2 levels) in parsing_binding'

```

can probably just do `build_version.to_s` - will need the same for below too

File: fastlane/Fastfile [hunk 8]
```diff
@@ -48,17 +48,55 @@ lane :ship_beta do
...
+                            sourcemap: "Pods/Emission/Pod/Assets/Emission.js.map",
+                            rewrite: true
+    puts "Uploaded Emission.js.map for #{project_slug}"
+
+    sentry_create_release auth_token: ENV['SentryUploadAuthKey'],
+                          org_slug: org_slug,
+                          project_slug: project_slug,
+                          version: latest_version, # TODO Does this create versions for the release number?
+                          finalize: true
+    puts "Created a release for #{project_slug}"
```
Comment (orta) [inline fastlane/Fastfile]: I wonder if this is order dependent, [last build failed due to a 404](https://circleci.com/api/v1.1/project/github/artsy/eigen/4169/output/22/0?file=true) - 

```
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api request POST https://sentry.io/api/0/projects/artsynet/eigen/releases/net.artsy.artsy-3.2.1/files/
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api using token authentication
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api sending form data
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api > POST /api/0/projects/artsynet/eigen/releases/net.artsy.artsy-3.2.1/files/ HTTP/1.1
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api > Host: sentry.io
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api > Accept: */*
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api > User-Agent: sentry-cli/1.9.2
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api > Authorization: Bearer 8539e384***
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api > Content-Length: 1103666
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api > Content-Type: multipart/form-data; boundary=------------------------dc05b4073d324bc9
DEBUG [2017-05-16 05:29:06.35]: [INFO] sentry_cli::api < HTTP/1.1 404 NOT FOUND
```

Perhaps moving this above the upload file etc

Or maybe make it above, then finalise it down here?

Comment (alloy) [inline fastlane/Fastfile]: Yup, done.

**Comment by alloy** [comment]
Any idea why the deploy setup tries to create 3.2.1-10 when that tag already exists?

**Comment by orta** [comment]
This _could be_ related to the fact that [we may already have successfully set the file](https://circleci.com/api/v1.1/project/github/artsy/eigen/4175/output/22/0?file=true)

```
DEBUG [2017-05-16 06:23:12.35]: [INFO] sentry_cli::api request POST https://sentry.io/api/0/projects/artsynet/eigen/releases/net.artsy.artsy-3.2.1/files/
DEBUG [2017-05-16 06:23:12.36]: [INFO] sentry_cli::api using token authentication
DEBUG [2017-05-16 06:23:12.36]: [INFO] sentry_cli::api sending form data

...

DEBUG [2017-05-16 06:23:12.36]: [INFO] sentry_cli::api response: 409
ERROR [2017-05-16 06:23:12.36]: error: File already present!
INFO [2017-05-16 06:23:12.95]: Sending crash report...
```

Right now, I guess we could wrap that in a try catch, but it looks like this file upload can only happen once per release? FWIW I don't see anything inside the website

screen shot 2017-05-16 at 14 31 39

**Comment by alloy** [comment]
The [generation of the version-build](https://github.com/artsy/eigen/blob/a100cfef203a56bbe5f027d0e457ba59cab5f7c8/fastlane/Fastfile#L44-L45) happens before anything to do with Sentry.

PS The reason you’re not seeing any files there is probably because you’re not looking at the staging version, which gets done first and if it fails there then nothing else happens.

**Comment by orta** [comment]
`3.2.1-10` means for version 3.2.1 there have been 9 TestFlight builds so far

screen shot 2017-05-16 at 14 34 13

As we keep failing to ship, it will keep trying with the same release

**Comment by alloy** [comment]
Aha. The tagging fails, though, so that means our tags might not have been in sync in the past either?

**Comment by orta** [comment]
It should be deleting and replacing, so no - shouldn't be:

```ruby
  tag = "#{latest_version}-#{build_version}"
  `git tag -d "#{tag}"`

 add_git_tag tag: tag
 `git remote add http https://github.com/artsy/eigen.git`
 `git push http #{tag}`

```

**Comment by alloy** [comment]
That only deletes a local tag, not on the remote (GitHub).

**Comment by orta** [comment]
it is pushed though, I wonder if that needs to a `-f`

**Comment by alloy** [comment]
Oh can you `-f` pushing tags? I always thought you needed to explicitly delete them from remote before pushing.

**Comment by orta** [comment]
Good q, will check the logs.

**Comment by orta** [comment]
Summary of where we are at:

* We want to deploy to our two Eigen projects in sentry: `Eigen` and `Eigen Staging`.
* We had issues shipping the dSYMs when they were a single zip file, but switching to unzipping ourselves and shipping one by one worked out fine.
* We started hitting 419 (`ERR_FILE_EXISTS`) issues on shipping the source for our React Native JS but only for the 2nd project being uploaded.

  * I had a look [at the source code](https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/project_release_files.py#L172-L183), and it _looks_ like when it creates a new file it is possible that it doesn't take the project into account.
  * Another, _unverified_, thought I had was that the the filename needs to be unique - as sentry was built for web projects, they would always have a unique URL for a file between projects e.g. `staging.x.com/file.js` vs `x.com/file.js`

While digging around, I've [shipped a successful beta](https://circleci.com/gh/artsy/eigen/4178) that _just_ sends the dysms/release/source-maps to `Eigen` aka production too.

**Comment by alloy** [comment]
Ok, that works for me. So for now I just need to perform another rebase on master and then cut a beta, right?

**Comment by orta** [comment]
Yah

**Comment by alloy** [comment]
💚

**Pull request merged**